storybookjs/storybook
# enable writing api documentation

Issue:

https://github.com/storybooks/storybook/issues/591#issuecomment-361617546

## What I did

I wanted to change what route storybook runs on and attempted to accomplish this with the following "full control" configuration.

```const path = require('path');
const genDefaultConfig = require('@storybook/react/dist/server/config/defaults/webpack.config.js');

module.exports = (baseConfig, env) => {
  const storybookBaseConfig = genDefaultConfig(baseConfig, env);

  storybookBaseConfig.entry.preview = storybookBaseConfig.entry.preview.map(function(entry) {
    if (entry.includes("webpack-hot-middleware")) {
      return `${require.resolve('webpack-hot-middleware/client')}?path=/book/__webpack_hmr&reload=true`;
    }

    return entry;
  });

  storybookBaseConfig.output.publicPath = '/book/';

  return storybookBaseConfig;
};
```

As mentioned in the issue comment I linked, the webpack hot middleware was not receiving the configuration it needed to enable a custom route.

I constructed a minimal configuration for webpack hot middleware including the `path`.


## How to test

Use a "full control" config as described and the default config and ensure HMR works for both storybooks

Is this testable with jest or storyshots?
Not sure

Does this need a new example in the kitchen sink apps?
I don't think so its not a common case.

Does this need an update to the documentation?
Possibly not since this is an edge case.

If your answer is yes to any of these, please make sure to include it in your PR.
